### DISCUSSION TITLE
Queries regarding airflow

### DISCUSSION THREAD

**Discussion by sandeshs11** [apache/airflow#31129] [Q&A] [answered]
Hello,

I am new to airflow and trying to understand this product. Have few questions. It will be good if anyone can answer those.

1. Suppose I have apache airflow installed on a server locally. In this case I would be able to run DAGs on same server. How can we run the DAGs on other servers? Is there Master-Agent configuration? Or we need to install airflow on all servers?
2. If Master-Agent configuration is there then is the communication secured? (SSL certificates?)
3. Is it possible to run DAG manually from backend (without using UI)? Do we need master in this case?
4. Airflow comes with inbuilt DB but documentation suggests to use external DB for production workloads. Is this really necessary? What are the risks of using in built DB?

**Comment by potiuk** [comment, marked as answer]
> Hello,
> 
> I am new to airflow and trying to understand this product. Have few questions. It will be good if anyone can answer those.
> 
> 1. Suppose I have apache airflow installed on a server locally. In this case I would be able to run DAGs on same server. How can we run the DAGs on other servers? Is there Master-Agent configuration? Or we need to install airflow on all servers?

Yes. you will find answers and opions in the Documentation: https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/overview.html

> 2. If Master-Agent configuration is there then is the communication secured? (SSL certificates?)

The architecture explains it. The components communicate via shared filesystem and shared database, so it depends how your DB/filesystem access is done. See https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/overview.html and more concepts there

> 3. Is it possible to run DAG manually from backend (without using UI)? Do we need master in this case?

Yes. you can use REST API for example (and scheduler and webserver need to be run for those, it will also use workers to execute them if you use different executors than Local executor). Details in the documenation https://airflow.apache.org/docs/apache-airflow/stable/stable-rest-api-ref.html

> 4. Airflow comes with inbuilt DB but documentation suggests to use external DB for production workloads. Is this really necessary? What are the risks of using in built DB?

Airflow has no in-built DB. The helm chart of ours (which is one of many deployment options) in it's basic configuration has a postgres DB embedded using bitnami subchart as a quick "development only" setup that allows you to try the helm chart quickly: https://airflow.apache.org/docs/helm-chart/stable/production-guide.html#database . Generally running production database means that you should have monitoring, backups, resilience to failure etc. Absolutely standard for any deployment of any database - and risks are the same as with any other application that uses database - i.e. data loss in case those things are not taken care about.

**Reply by sandeshs11** [reply]
Thank you for the quick response.

I checked Documentation: https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/overview.html

|| An executor, which handles running tasks. In the default Airflow installation, this runs everything inside the scheduler, but most production-suitable executors actually push task execution out to workers.

So you are referring to below executors right? 

Remote Executors

[Celery Executor](https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/executor/celery.html)
[CeleryKubernetes Executor](https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/executor/celery_kubernetes.html)
[Dask Executor](https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/executor/dask.html)
[Kubernetes Executor](https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/executor/kubernetes.html)
[LocalKubernetes Executor](https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/executor/local_kubernetes.html)

These need to be setup in case workload need to be executed on remote machines?

**Reply by potiuk** [reply]
Yes
